utils/survival_cache_integration.py

```
@@ -0,0 +1,153 @@
+    
+    Args:
+        calculate_func: Function that calculates KM curves
+        cache_key_params: Dict with parameters for cache key
+    
+    Returns:
+        KM curve data (from cache or fresh calculation)
+    """
+    # Try cache first
+    cached = COMPUTATION_CACHE.get('survival_km', **cache_key_params)
+    if cached is not None:
+        logger.info(f"✅ Survival KM Cache HIT - using cached curves")
```

Ruff (0.14.10)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)